olp-cpp-sdk-core/cmake/curl.cmake

```
@@ -40,14 +40,14 @@ if(CURL_FOUND AND NOT NETWORK_NO_CURL)
         add_definitions(-DENABLE_CURL_VERBOSE)
     endif()
 
-    include(CheckFunctionExists)
+    include(CheckSymbolExists)
     add_definitions(-DNETWORK_HAS_UNISTD_H=1)
-    check_function_exists("pipe" HAS_PIPE)
-    if(HAS_PIPE)
+    check_symbol_exists(pipe unistd.h OLP_SDK_HAS_PIPE)
```

```suggestion
    check_symbol_exists(pipe "unistd.h" OLP_SDK_HAS_PIPE)
```

As specified in the official cmake documentation, https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html.